ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/StoreBackedCommandExecutor.scala

```diff
@@ -46,19 +46,20 @@ private[apiserver] final class StoreBackedCommandExecutor(
       loggingContext: LoggingContext,
   ): Future[Either[ErrorCause, CommandExecutionResult]] = {
     val start = System.nanoTime()
+    val actAsReadAsUnion = commands.actAs ++ commands.readAs
     val submissionResult = Timed.trackedValue(
       metrics.daml.execution.engineRunning,
-      engine.submit(commands.commands, participant, submissionSeed))
-    consume(commands.submitter, submissionResult)
+      engine.submit(actAsReadAsUnion, commands.commands, participant, submissionSeed))
```

Comment:
```suggestion
      engine.submit(commands.actAs, commands.commands, participant, submissionSeed))
```

The engine should only know about the `actAs` parties. The `readAs` parties are part of the visibility check that is done outside of the engine. Maybe rename the `actAsReadAsUnion` value to make it more obvious:
```scala
// The actAs and readAs parties are used for two kinds of checks by the ledger API server:
// When looking up contracts during command interpretation, the engine should only see contracts
// that are visible to at least one of the actAs or readAs parties. This visibility check is not part of the
// DAML ledger model.
// When checking DAML authorization rules, the engine verifies that the actAs parties are sufficient to
// authorize the resulting transaction.
val contractReaders = commands.actAs ++ commands.readAs
val commitAuthorizers = commands.actAs

engine.submit(commitAuthorizers, ...)
consume(contractReaders, ...)
```

Note that this error would be caught by end-to-end tests for multi-party submissions, which we won't have until the PR that enables multi-party submissions.

Reply:
Thank you for this explanation!

Reply:
Fixed.